Heteroatom database updates and additions

alongd opened the pull request:
* NOx2018 kinetic, thermo, and transport libraries were added from P. Glarborg, J.A. Miller, B. Ruscic, S.J. Klippenstein 2018 (https://doi.org/10.1016/j.pecs.2018.01.002)
* Refactored the kinetic library HSSH_1bar 
* The NOx and SOx kinetic libraries were renamed to PrimaryNitrogenLibrary and PrimarySulfurLibrary, since they deal with a broader range of chemistry than only NOx and SOx, and to avoid confuition with the NOx2018 library.
*  Added the kinetic N-S_interactions library, a spin-off of the PrimarySulfurLibrary. This separation is important for cases when a sulfur system simulation with nonreactive N2 is desired.
* Added Green2014 and HCCO+NO reactions to the PrimaryNitrogenLibrary
*  Added the SulfurHaynes thermo library for H2S oxidation (doi: 10.1016/j.proci.2012.05.083)
* Improvements and additions to the PrimarySulfurLibrary
*  Additions to the SOxNOx thermo library at the CCSD(T)-F12a/cc-pVTZ-f12//M06-2x/cc-pVTZ level of theory (O2ScycOO, SO2(T), OOS(T), HSOS, cycOOS, cycOOSS, and NNS). These are good candidates for GAV elaboration as they contain birads and cyclics.
* Globally forbidding S2O2, SOO2, SO2O2, N2SH , which were predicted by RMG, yet their geometry could not be optimized with realistic bond lengths (they are probably not minima). When the forbiddedStructure method improves, these could be forbidden as species, not structures.
* Other minor corrections, inc. some training reactions

mjohnson541 commented on input/kinetics/families/H_Abstraction/training/reactions.py:
```diff
@@ -2172,7 +2172,7 @@
     kinetics = Arrhenius(A=(1.48e+22, 'cm^3/(mol*s)'), n=-3.37, Ea=(5429, 'cal/mol'), T0=(1, 'K'), Tmin = (300, 'K'), Tmax = (3000, 'K')),
     rank = 2,
-    shortDesc = u"""Training reaction from kinetics library: SOxNOx""",
+    shortDesc = u"""Training reaction from kinetics library: NOx""",
```
Note these changes in the commit

mjohnson541 commented on input/kinetics/families/R_Recombination/training/reactions.py:
```diff
@@ -203,7 +203,7 @@
     kinetics = Arrhenius(A=(5.88e+24, 's^-1'), n=-2.35, Ea=(62398, 'cal/mol'), T0=(1, 'K'), Tmin = (500, 'K'), Tmax = (3000, 'K')),
     rank = 2,
-    shortDesc = u"""Training reaction from kinetics library: SOxNOx""",
+    shortDesc = u"""Training reaction from kinetics library: NOx""",
```
Note this change in the commit

mjohnson541 commented on input/kinetics/libraries/Sulfur/HSSH_1bar/reactions.py:
```diff
@@ -0,0 +1,86 @@
+[Sendt2009b] C.R. Zhou, K. Sendt, B.S. Haynes, J. Phys. Chem. A, 2009, 113, 8299-8306, doi: 10.1021/jp903185k
+"""
+
+entry(
```
Clearly you've changed the composition of the library significantly, can you note that in the commit message?

alongd replied:
Done

alongd commented on input/kinetics/libraries/SOx/reactions.py:
```diff
@@ -1134,140 +1119,264 @@
+    shortDesc = u"""[Sendt2009b]""",
+    longDesc =
+u"""
+Table 1, R3
```
Do you think `Table 1, reaction R3` or `Reaction R3 in Table 1` is clearer? I used that notation (without explicitly writing `reaction`) throughout several libraries. I think that if someone opens the referenced work, and looks for Table 1, then the current notation is pretty much self explanatory, but I don't mind changing it if you think otherwise.

mjohnson541 replied:
Oh, this is a reference within the paper?

mjohnson541 replied:
Ok, I guess Table 1 R3 is ok

alongd commented on input/forbiddenStructures.py:
```diff
@@ -531,3 +531,163 @@
 """,
 )
+
+entry(
```
Yes! I'll make these changes

alongd commented on input/kinetics/libraries/Sulfur/GlarborgH2S/reactions.py:
```diff
@@ -795,6 +794,27 @@
 """,
 )
 
+# entry(
```
I'm addressing other comments in fixup commits, but reworded this already existing commit

alongd commented on input/kinetics/libraries/Sulfur/GlarborgMarshall/reactions.py:
```diff
@@ -1752,6 +1752,7 @@
     longDesc = 
 u"""
 ALZ/GLA01 HIG/MUR80
+alongd comment: HIG/MUR80 is https://doi.org/10.1246/bcsj.53.15; Rate determined at 4500-6000 K
```
Yes...
```
The disappearance rate of S2, which was produced from the pyrolysis of COS, was measured behind
incident shock waves by monitoring the UV emission in the temperature range of 4500–6000 K and in
the pressure range of 0.32–0.5 atm. It was found that two processes could possibly account for the
disappearance of S2:
```

I'm using this high T data point in the PrimarySulfurLibrary for the same reaction to give a better rate estimation at reasonable Ts. I though I'll drop a line here so people know...

alongd commented:
Thanks for reviewing! Rebased, corrected commit messages, and added a fixup commit for forbiddenStructures to be squashed before merging

mjohnson541 commented on input/transport/libraries/NOx2018.py:
```diff
@@ -0,0 +1,4351 @@
+    7 O u0 p2 c0 {2,D}
+    8 O u0 p3 c-1 {2,S}
+    """,
+    transport=Transpor
```
So this a set of comments describing the original chemkin transport file and explaining what you excluded from the RMG transport library?

alongd replied:
Yes, sort of.
The first sentence is my addition, the rest is simply the original transport file.
I added this because (a) not all comments were parsed correctly, (b) this library includes other species which weren't mentioned in the NOx2018 thermo and kinetic libraries. I did not have structures for most of these additional species, so they were excluded. I'd happily add this to the comment to explain why they weren't added.

mjohnson541 replied:
Is there some reason we need to include the entire file in the library rather than just explain that we only added species that were in the NOx2018 library, but more transport data is available in such and such paper?

alongd replied:
I thought it would be more convenient to look at the unparsed comments this was. No problem removing it if you prefer.

mjohnson541 replied:
I mean I understand leaving it in if we've processed/commented on the information somehow, or the information is difficult to access, but if it's just a matter of looking up the paper I feel like we should just comment that we only included species in the NOx2018 library and that there are more in the paper.

mjohnson541 commented on input/kinetics/libraries/SOx/reactions.py:
```diff
@@ -832,11 +849,11 @@
     degeneracy = 1,
     kinetics = ThirdBody(
-        arrheniusLow = Arrhenius(A=(1.60e+24, 'cm^6/(mol^2*s)'), n=-2.613, Ea=(89173, 'cal/mol'), T0 = (1, 'K'),
+        arrheniusLow = Arrhenius(A=(1.15e+25, 'cm^6/(mol^2*s)'), n=-2.840, Ea=(1665, 'cal/mol'), T0 = (1, 'K'),
```
Was this resolved?

alongd replied:
No, thanks for reminding me. I'll check it again

mjohnson541 replied:
Was this resolved?

alongd replied:
Yes - it was a copy-past error, I added a note in the commit message

alongd commented:
@mjohnson541, last comments were addressed: corrected commit messages to reflect all changes and added fixup commits for forbidden structures and NOx2018 transport library. Let me know whether you have additional comments.

alongd commented:
Thanks, @mjohnson541! Fixup commits were squashed, ready to be merged once tests pass

alongd commented:
Corrected the H_Abstraction family-specific forbidden group "disprop1" so `HO2` isn't forbidden.
We'll wait for the tests before merging this in

alongd commented:
RMG-tests mostly make sense now.

In eg1 there's non-identical thermo with a slight change in entropy for this edge species (I don't know why, but it's very minor):
```
original:	[CH]1CCC1CC1CC1
tested:	[CH]1CCC1CC1CC1
Hf(300K)  |S(300K)   |Cp(300K)  |Cp(400K)  |Cp(500K)  |Cp(600K)  |Cp(800K)  |Cp(1000K) |Cp(1500K) 
     65.21|     98.92|     32.87|     44.26|     54.81|     63.99|     77.53|     85.89|    100.40
     65.21|     97.54|     32.87|     44.26|     54.81|     63.99|     77.53|     85.89|    100.40
Identical thermo comments
```

The NC test is missing some ~40 edge reactions, all H Abstraction. These make sense -- the H is being abstracted from a site adjacent to a radical, this is now forbidden and should only be generated as a disprop reaction.

Some other thermo entries are slightly different when ring corrections for `oxirene` and `Cyclopropene` (or similar) interchange.  e.g.:
```
Non-identical thermo!
original:	O=C1CO1
tested:	O=C1CO1
Hf(300K)  |S(300K)   |Cp(300K)  |Cp(400K)  |Cp(500K)  |Cp(600K)  |Cp(800K)  |Cp(1000K) |Cp(1500K) 
    -36.59|     59.67|     11.69|     15.39|     18.97|     21.97|     25.65|     27.25|     31.22
    -41.42|     63.49|     13.90|     16.17|     18.31|     20.29|     23.68|     26.27|     29.84
thermo: Thermo group additivity estimation: group(Cs-(Cds-O2d)OsHH) + group(Cds-OdCsOs) + group(O2s-Cs(Cds-O2d)) + ring(cyclopropanone)
thermo: Thermo group additivity estimation: group(Cs-(Cds-O2d)OsHH) + group(Cds-OdCsOs) + group(O2s-Cs(Cds-O2d)) + ring(2(co)oxirane)
```

There are several cases of non-identical kinetics involving small cyclics which thermo is slightly different as described above, probably because of the change in DeltaHrxn (indeed only the Ea was changed)

The missing H Abstraction reactions in the NC test are a desired change, all other changes seem minor. @mjohnson541, do you think this could be merged?

[main_log.txt](https://github.com/ReactionMechanismGenerator/RMG-database/files/1827467/main_log.txt)

mjohnson541 commented:
Ok, I think this is good to merge!

alongd commented:
The last fixup commit organizes the disprop1 forbidden groups in H_abstraction.
Generally, we'd like to forbid `HR[R].` from reacting here (`.` marks a radical), since this should be a disprop reaction. However, the following specific cases must not be forbidden here: `HO2`, `HSS`, `HOS`, `HSO` (since they form the ground state triplets O2, S2, and SO).
Hence, the solution might be to forbid the following groups:
- `HR[C,N].` the radical site isn't O or S
- `H[C,N][O,S].` the radical site is O or S, but the non-rad site isn't
- `H[S p0,1][O,S].` hypervalance S is allowed at the H site
- `H[O,S][S p0,1].`hypervalance S is allowed at the rad site

@mjohnson541, I think this is the last fix for this PR, please let me know if this looks OK to you. I'd also appreciate any feedback from you and @mliu49 on whether this could be made more efficient, since H_Abs is used frequently.

alongd commented:
@mjohnson541, I added fixup commits with general changes related to this branch. Specifically, I'd like your feedback on the H_abs forbidden group change.

mjohnson541 commented:
I'm at a meeting right now, but I'll try to take a look at it tonight.
